.github/pull_request_template.md

```diff
@@ -1,62 +1,83 @@
+  blank, we will help you out during the review.
+-->
 
-1. [ ] I updated [`ChangeLog.md`](https://github.com/wasp-lang/wasp/blob/main/waspc/ChangeLog.md) with description of the change this PR introduces.
-2. [ ] I bumped `waspc` version in [`waspc.cabal`](https://github.com/wasp-lang/wasp/blob/main/waspc/waspc.cabal) to reflect changes I introduced, with regards to the version of the latest wasp release, if the bump was needed.
-
-### Add a regression test if needed
+### Description
 
-If you did a **bug fix**, make sure you satisfy the following:
+<!--
+  Describe your PR!
 
-1. [ ] I added a regression test that reproduces the bug and verifies the fix.
+  Common questions we'd like you to answer:
+  - What's the motivation for this change?
+  - Which changes are included in this PR?
+    - If there are many different changes, consider splitting your PR into smaller
+      ones. It will go through faster!
 
-If you're unable to add a regression test, please explain why.
-This likely indicates that our current testing setup needs improvement.
+  If this PR closes an issue, write “Fixes #XXXX" so GitHub will link them together.
 
-### Test Coverage
+  You can also answer some of these questions if they are relevant:
+  - Does this change affect users? How?
+  - Have you considered any other approaches? Why is this one the best?
+  - Are there any drawbacks or edge cases?
+  - What are the possibilities for future work?
+-->
 
-Please ensure your changes are adequately tested:
+### Select what type of change this PR introduces:
 
-1. [ ] **My changes are covered by tests** (unit, integration, or e2e tests as appropriate).
+<!-- Put an x in between the brackets to select options, like so: [x] -->
 
-If you're unable to add tests or if coverage is partial, please explain why below:
+- [ ] **Just code/docs improvement** (no functional change).
+- [ ] **Bug fix** (non-breaking change which fixes an issue).
+- [ ] **New feature** (non-breaking change which adds functionality).
+- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).
 
-<!-- Provide explanation here if tests are missing or incomplete -->
+### Checklist
 
-### Update example apps if needed
+<!-- Put an x in between the brackets to select options, like so: [x] -->
+<!-- You can add notes or explanations wherever needed -->
 
-If you did code changes and **added a new feature**, make sure you satisfy the following:
+- 🧪 Testing:
 
-1. [ ] I updated [`waspc/examples/todoApp`](https://github.com/wasp-lang/wasp/tree/main/waspc/examples/todoApp) and its e2e tests as needed and manually checked it works correctly.
+  - [ ] I tested this change in a Wasp app **locally**.
+  - [ ] <!-- If you modified Haskell code: --> I added **unit tests** for my change at `waspc/tests`.
+  - [ ] <!-- If you added or updated a feature: --> I added an **integration test** for my change at `waspc/examples/todoApp/e2e-tests`.
```

Comment:
I would call these e2e tests, to not introduce another term, they are called e2e tests anyway.